nomad/fsm_test.go

```diff
@@ -3422,3 +3423,61 @@ func TestFSM_ACLEvents(t *testing.T) {
 		})
 	}
 }
+
+// TestFSM_EventBroker_JobRegisterFSMEvents asserts that only a single job
+// register event is emitted when registering a job
+func TestFSM_EventBroker_JobRegisterFSMEvents(t *testing.T) {
+	t.Parallel()
+	fsm := testFSM(t)
+
+	job := mock.Job()
+	eval := mock.Eval()
+	eval.JobID = job.ID
+
+	req := structs.JobRegisterRequest{
+		Job:  job,
+		Eval: eval,
+	}
+	buf, err := structs.Encode(structs.JobRegisterRequestType, req)
+	if err != nil {
```

Comment:
You can do `require.NoError(t, err)` for this block

Reply:
woops :)